src/org/traccar/events/OverspeedEventHandler.java

```diff
@@ -50,8 +55,48 @@ public OverspeedEventHandler() {
 
         Collection<Event> events = new ArrayList<>();
         double speed = position.getSpeed();
+        double speedLimit = 0;
 
-        if (globalSpeedLimit != 0 && speed > globalSpeedLimit) {
+        if (device.getAttributes().containsKey(ATTRIBUTE_SPEED_LIMIT)) {
+            speedLimit = Double.parseDouble((String) device.getAttributes().get(ATTRIBUTE_SPEED_LIMIT));
+        }
+        if (speedLimit == 0 && checkGroups) {
```

Comment:
Should we move attribute lookup logic somewhere else. It will be reused for other attributes as well, right?